logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/folder/ObserveUserFoldersUseCase.kt

```diff
@@ -39,7 +41,15 @@ internal class ObserveUserFoldersUseCaseImpl(
 ) : ObserveUserFoldersUseCase {
 
     override suspend operator fun invoke(): Flow<List<ConversationFolder>> {
-        return conversationFolderRepository.observeUserFolders()
+        return conversationFolderRepository.observeFolders()
+            .mapRight { folders ->
+                if (folders.isEmpty()) {
+                    conversationFolderRepository.fetchConversationFolders()
+                    emptyList()
```

Comment:
question: should fetching happen every time when the app observes and there is no folder? And why does it return empty list after fetching - what if `fetchConversationFolders` actually returns some folders but here it still returns empty list 🤔

Reply:
Fetch folders will automatically create favorite folder when there is none on backend, so this will not fetch them every time